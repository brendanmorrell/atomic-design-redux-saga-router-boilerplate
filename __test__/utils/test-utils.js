import React from 'react';
import { render } from 'react-testing-library';
import { ThemeProvider } from 'styled-components';
import theme from '../../src/theme';
import { Router } from 'react-router';
import { Provider } from 'react-redux';
import { storeFactory, history } from '../../src/store';

export const store = storeFactory({});
const customRender = (node, options) =>
  render(
    <Provider store={store}>
      <Router history={history}>
        <ThemeProvider theme={theme}>{node}</ThemeProvider>
      </Router>
    </Provider>,
    options,
  );

// re-export everything
export * from 'react-testing-library';

// override render method
export { customRender as render };
