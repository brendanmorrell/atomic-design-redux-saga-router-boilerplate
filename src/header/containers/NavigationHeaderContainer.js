import React from 'react';
import styled from 'styled-components';
import { breakpoint } from '../../theme/Mixins';
import { connect } from 'react-redux';

import { navigate } from '../../sidebar';
import { showMobileMenu } from '../../mobilemenu';
import { history } from '../../store';

import LogoSrc from '../../assets/logo.png';
import MobileMenuSrc from '../../assets/icons/mobile-menu.svg';

import NavigationBar from '../molecules/NavigationBar';
import AuthSidebar from '../../auth/containers/AuthSidebar';
import ProfileSidebar from '../../profile/containers/ProfileSidebar';

const Container = styled.div`
  position: relative;
  width: 100vw;
  height: 75px;
  background-color: #fff;
  border-bottom: 1px solid #efefef;
`;

const Content = styled.div`
  position: relative;
  width: 100%;
  max-width: 1440px;
  margin: 0 auto;
  height: 75px;
  transition: all 0.25s;
`;

const Logo = styled.img`
  position: absolute;
  width: 175px;
  left: 50px;
  top: 50%;
  transform: translateY(-50%);
  transition: all 0.2s;
  ${breakpoint.down('m')`
    left: 20px;
  `}
`;

const Auth = styled.div`
  position: absolute;
  top: 50%;
  right: 40px;
  transform: translateY(-50%);
  cursor: pointer;
`;

const MobileMenu = styled.div`
  position: absolute;
  top: 50%;
  right: 190px;
  transform: translateY(-50%);
  cursor: pointer;
  display: none;
  ${breakpoint.down('m')`
    display: inline;
  `}
`;

const NavigationHeaderContainer = ({
  navigateAction,
  showMobileMenuAction,
}) => {
  return (
    <>
      <Container>
        <Content>
          <Logo
            data-testid={'/'}
            src={LogoSrc}
            onClick={() => {
              history.push('/');
            }}
          />

          <NavigationBar />

          <Auth
            onClick={() => {
              navigateAction({
                component: <AuthSidebar />,
              });
            }}
          >
            Sign Up
          </Auth>

          <Auth
            style={{ right: 120 }}
            onClick={() => {
              navigateAction({
                component: <ProfileSidebar />,
              });
            }}
          >
            Profile
          </Auth>

          <MobileMenu onClick={showMobileMenuAction}>
            <img src={MobileMenuSrc} alt="Menu" />
          </MobileMenu>
        </Content>
      </Container>
    </>
  );
};

const mdtp = {
  navigateAction: navigate,
  showMobileMenuAction: showMobileMenu,
};

export default connect(
  null,
  mdtp,
)(NavigationHeaderContainer);
