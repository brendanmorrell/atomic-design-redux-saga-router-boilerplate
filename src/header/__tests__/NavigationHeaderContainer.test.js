import React from 'react';
import { render, fireEvent, debugDOM, flushPromises } from 'test-utils';
import NavigationHeaderContainer from '../containers/NavigationHeaderContainer';

test('snapshot', () => {
  const { container } = render(<NavigationHeaderContainer />);
  expect(container.firstChild).toMatchSnapshot();
});

test('should render all routes', () => {
  const { getByText } = render(<NavigationHeaderContainer />);
  // const shopLink = getByText('Shop');
  // console.log(debugDOM(document.body));
  // fireEvent.click(shopLink);
  // expect(window.location.pathname).toBe('');
});

// { title: 'Shop', url: '/shop', menu: null },
// {
//   title: 'Our Brands',
//   url: null,
//   menu: [{ title: '[statemade]', url: '/brands' }],
// },
// { title: 'Apparel', url: '/apparel', menu: null },
// { title: 'EMBER', url: '/blog', menu: null },
// { title: 'Investors', url: '/investors', menu: null },
