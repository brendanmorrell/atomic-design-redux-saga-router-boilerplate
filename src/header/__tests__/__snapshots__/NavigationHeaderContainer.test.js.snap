// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  class="NavigationHeaderContainer__Container-sc-15wz8bx-0 bfEDXE"
>
  <div
    class="NavigationHeaderContainer__Content-sc-15wz8bx-1 eKEtxa"
  >
    <img
      class="NavigationHeaderContainer__Logo-sc-15wz8bx-2 gRjkhR"
      src="logo.png"
    />
    <div
      class="NavigationBar__Container-sc-1mqwq84-0 LeeJg"
    >
      <div
        class="NavigationLink__Container-zdydky-0 jIDNwV"
      >
        <div
          class="NavigationLink__Title-zdydky-1 jubtma"
        >
          <a
            href="/shop"
          >
            Shop
          </a>
        </div>
      </div>
      <div
        class="NavigationLink__Container-zdydky-0 jIDNwV"
      >
        <div
          class="NavigationLink__Title-zdydky-1 jubtma"
        >
          <span>
            Our Brands
          </span>
        </div>
      </div>
      <div
        class="NavigationLink__Container-zdydky-0 jIDNwV"
      >
        <div
          class="NavigationLink__Title-zdydky-1 jubtma"
        >
          <a
            href="/apparel"
          >
            Apparel
          </a>
        </div>
      </div>
      <div
        class="NavigationLink__Container-zdydky-0 jIDNwV"
      >
        <div
          class="NavigationLink__Title-zdydky-1 jubtma"
        >
          <a
            href="/blog"
          >
            EMBER
          </a>
        </div>
      </div>
      <div
        class="NavigationLink__Container-zdydky-0 jIDNwV"
      >
        <div
          class="NavigationLink__Title-zdydky-1 jubtma"
        >
          <a
            href="/investors"
          >
            Investors
          </a>
        </div>
      </div>
    </div>
    <div
      class="NavigationBar__Overlay-sc-1mqwq84-1 jzhFpb"
    />
    <div
      class="NavigationHeaderContainer__Auth-sc-15wz8bx-3 hCCDtr"
    >
      Sign Up
    </div>
    <div
      class="NavigationHeaderContainer__Auth-sc-15wz8bx-3 hCCDtr"
      style="right: 120px;"
    >
      Profile
    </div>
    <div
      class="NavigationHeaderContainer__MobileMenu-sc-15wz8bx-4 kPVmxG"
    >
      <img
        alt="Menu"
        src="mobile-menu.svg"
      />
    </div>
  </div>
</div>
`;
